pkg/workflow/workflow.go

```diff
@@ -0,0 +1,127 @@
+package workflow
+
+import (
+	"context"
+)
+
+// Config is the configuration for constructing a Workflow instance.
+type Config struct {
+	// ErrorHandler is handler called when a workflow experiences an error. The error may originate
+	// from hook or from a task. The original error is alwasy returned from the workflow's Execute.
+	// Optional. Defaults to a no-op handler.
+	ErrorHandler ErrorHandler
```

Comment:
I would expect the error handler to define those rules. We would need to provide information on the current task and previous tasks which shouldn't be difficult via the context. That might even be a good justification for a custom context given it will be applicable to everything.

Reply:
Ah I see. Like the we would have an error handler to orchestrates other handlers, sounds good.

If we know we need to pass that kind info (tasks executed, etc.) to the handler, doesn't that guarantee a dedicated argument for that?

Reply:
Possibly. I've purposely left this rather plain for this PR. I included it so we have an appreciation for how this could be done. I have an open issue to actually implement this including the design work/pocing.